src/ocean/io/serialize/StringStructSerializer.d

```diff
@@ -845,4 +846,30 @@ unittest
                          "   ulong a : 1000\n",
             "Incorrect string serializer result");
     }
+
+    // test a typedef array
+
+    struct StructWithTypedefArray
+    {
+        AdskilletId[] ids;
+    }
+
+    StructWithTypedefArray sta;
+
+    sta.ids = new AdskilletId[](4);
+
+    for (int idx = 0; idx < 4; idx++) 
```

Comment:
Foreach ref?

Reply:
isn't the for loop is clearer ?

Reply:
I can a hard time imagining situation when it could be clearer. It is also more error-prone.